Add tutorial for basic ua commands (SC-1056)

lucasmoura opened the pull request:
## Proposed Commit Message
Add tutorial for basic ua commands

## Test Steps
Run the tutorial and see if the flow makes sense and everything works as expected

## Desired commit type
<!-- put an `x` in one merge type to allow the reviewer to merge for you -->
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes
that apply. -->
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

CalvoM commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,257 @@
+### Attach
+
+To access any of those service offerings, the user need to attach UA to a subscription. This is
+achieved by running the attach command. Before we run it, we need to get ourselver a token.
```
Minor change - ourselves.

CalvoM commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,257 @@
+$ sudo ua refresh
+```
+
+This command will refresh the contract on your machine. This command is also really useful if you
````
Could we add the fact that `ua refresh` also updates the config esp. since `ua refresh --help` says that 
```sh
`ua refresh` is the equivalent of `ua refresh config && ua refresh contract`.
```

lucasmoura replied:
That's a good point. I have not added more context about refresh here because I have not discussed in the tutorial about the config file or even MOTD messages. I think we can add a note at the end saying that refresh perform more actions on the machine and link this into an explanation doc. What do you all think ? @CalvoM @renanrodrigo @orndorffgrant

renanrodrigo replied:
just a text line here would do imho, mentioning it will also update UA configuration. When we have a Config tutorial, we can add a link here pointing to it.

lucasmoura replied:
Okay, I will put another doc explaining what refresh does in the system and then I will reflect it on this section

CalvoM commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,257 @@
+After running it. you should see an output similar to this one:
+
+```
+ne moment, checking your subscription first
````
Really minor change, `One moment`

renanrodrigo commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,257 @@
+
+### Attach
+
+To access any of those service offerings, the user need to attach UA to a subscription. This is
```
```suggestion
To access any of those service offerings, the user needs to attach UA to a subscription. This is
```
or, as we are being informal,
```suggestion
To access any of those service offerings, the you need to attach UA to a subscription. This is
```

renanrodrigo commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,257 @@
+One question that may arise is that the output of status and unattached are different. When
+attached, status presents two new columns, **ENTITLED** and **STATUS**, while also dropping the
+**AVAILABLE** column. For more information of why the output is different, please refer to this
+[explanation](../explanations/status_columns.md)
```
```suggestion
[explanation](../explanations/status_columns.md).
```

renanrodrigo commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,257 @@
+Valid until: 9999-12-31 00:00:00+00:00
+```
+
+The `Valid until` field describe when your contract will be expired, so you can be aware of when it
````
```suggestion
The `Valid until` field describes when your contract will be expired, so you can be aware of when it
```

renanrodrigo commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,257 @@
+$ sudo ua enable cis
+```
+
+After running it. you should see an output similar to this one:
````
```suggestion
After running it you should see an output similar to this one:
```

renanrodrigo commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,257 @@
+the tool, please take a look on [How to guides](./docs/howtoguides). If that still doesn't cover
+your needs, feel free to reach the UA team on `#ubuntu-server` on Libera IRC.
+
+Before we finish this tutorial, let's just delete our LXD container by running:
```
Maybe it is useful to tell the user to run this on the machine, and not inside of the container where the previous commands were executed.

renanrodrigo reviewed (approved):
Found some grammar things, but I am no native English speaker so, well, not sure.

It is not wrong or anything, but personally I think we could standardize the pronouns across the tutorial. I don't think there is need to make it formal, getting rid of "we" or "you", but I think it is more readable when we always use the same pronoun.
Like in:
```
Finally, what if I don't want to be attached to an UA subscription any longer 
We can achieve that through the detach command:
(...)
This command will disable all of the UA services on the machine for you
```
could be 
```
Finally, what if we don't want to be attached to an UA subscription any longer 
We can achieve that through the detach command:
(...)
This command will disable all of the UA services on the machine for us
```
The same works for I, or you, or we.

This is just a suggestion. Feel free to keep as is if you think that is too pedantic.

lucasmoura commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,257 @@
+## Prerequisites
+
+On this tutorial, we will use [LXD](https://linuxcontainers.org/lxd/) containers.
+To set LXD on your computer, please follow this [guide](https://linuxcontainers.org/lxd/getting-started-cli/).
```
I think it is `set up LXD`, but let's @orndorffgrant make the call here haha

orndorffgrant replied:
`set up LXD` makes the most sense to me :+1:

lucasmoura commented:
@renanrodrigo @CalvoM I have addressed the comments you made.
@renanrodrigo I have used your suggestion and I am now using `you` as the standard pronoun through the tutorial. Let me know If I missed anything

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+# Tutorial: cover base UA commands
+
+The Ubuntu Advantage (UA) Tools provides users with a simple mechanism to
+view, enable, and disable offerings from Canonical on their system. On this tutorial
```
```suggestion
view, enable, and disable offerings from Canonical on their system. In this tutorial
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+See https://ubuntu.com/advantage
+```
+
+You can see that the status command shows the service that are available for that given machine,
````
```suggestion
You can see that the status command shows the services that are available for that given machine,
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+
+### Attach
+
+To access any of those service offerings, you need to attach UA to a subscription. This is
```
I suggest either
```suggestion
To access any of those service offerings, you need to attach `ua` to a subscription. This is
```
or
```suggestion
To access any of those service offerings, you need to attach to a UA subscription. This is
```

lucasmoura replied:
Going with the second one

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+```
+
+From this output, you can see that the attach command enables all of the services specified by the user subscription.
+After the command ends, UA display what is the state of the machine now.
````
```suggestion
After the command ends, `ua` displays the new state of the machine.
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+$ ua status
+```
+
+One question that may arise is that the output of status and unattached are different. When
````
```suggestion
One question that may arise is that the output of `ua status` while attached is different from the output of `ua status` when unattached. When
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+
+### Refresh
+
+In the last section, we mentioned that your contract can expire. After you renew the contract
+how can you make our machine aware of that ? You can do this through this command:
```
We should include some assurance here that free tokens do not expire.

We can say something like "While free tokens never expire, if you buy a UA subscription, and later need to renew the contract, you can make your machine aware of it by running this command."

lucasmoura replied:
Modified it a little bit, but I agree it is information worth having on the tutorial

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+### Enable
+
+There is another way to enable a service that wasn't activated during attach or refresh.
+Suppose that you don't want to modify your subscription, but still want to enable `cis`
+on the machine. To achieve that, You can use the enable command.
```
Simplification suggestion :shrug: 
```suggestion
Suppose that you want to enable `cis` on this machine manually. To achieve that, You can use the enable command.
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+$ sudo ua disable cis
+```
+
+After running that command, let's now run status to see what happened to `cis`:
````
```suggestion
After running that command, let's now run `ua status` to see what happened to `cis`:
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+
+You can see that `cis` status is back to disabled.
+
+```
+Note: the disable command doesn't uninstall any package that was installed by
+the service. The command only removes the access you have to the service, but it
+doesn't undo any configuration that was applied on the machine.
+```
````
Suggest blockquotes for notes like these
```suggestion
> Note: the disable command doesn't uninstall any package that was installed by
> the service. The command only removes the access you have to the service, but it
> doesn't undo any configuration that was applied on the machine.
```

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+
+### Detach
+
+Finally, what if I don't want to be attached to an UA subscription any longer ?
+You can achieve that through the `detach` command:
```
I haven't paid close enough attention to this, but we should stick with one "subject" throughout the tutorial. In other words, we should pick one of "I", "We", or "You" and use it consistently. I think we've mostly been using "We", so lets keep that going here.

Also some minor things

```suggestion
Finally, what if we don't want this machine to be attached to a UA subscription any longer?
We can achieve that using the `detach` command:
```

lucasmoura replied:
That's a good point. I am trying to use `you` in this doc. The places were I am using `we` are places were I mentioning that the UA team will perform an action. But let me know if  ẁe` is better instead of `you` on the docs

orndorffgrant replied:
I somehow didn't see @renanrodrigo's comment that says pretty much the exact same thing :facepalm: 

I think `you` is good too. And `we` would make sense in those situation :+1:

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
````diff
@@ -0,0 +1,258 @@
+```
+Note: the detach command will also not uninstall any packages that were installed by
+any service enabled through UA.
+```
````
I suggest blockquote here as well

orndorffgrant commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+### Final thoughts
+
+This tutorial has covered the 6 main commands of UA. If you need more advanced options to configure
+the tool, please take a look on [How to guides](./docs/howtoguides). If that still doesn't cover
```
```suggestion
the tool, please take a look in [How to guides](./docs/howtoguides). If that still doesn't cover
```

lucasmoura commented on README.md:
```diff
@@ -52,13 +52,15 @@ Users can manually run the `ua` command to learn more or view the manpage.
 ### Tutorials
 
 * [Create a FIPS compliant Ubuntu Docker image](./docs/tutorials/create_a_fips_docker_image.md)
+* [UA basic commands](./docs/tutorials/basic_ua_commands.md)
```
I agree it should be the first one, but what about just `Getting started with UA` ?
I think this is generic enough so we can even increment that tutorial in the future while also being a more inviting title

orndorffgrant replied:
Good idea! I like it

lucasmoura commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+# Tutorial: cover base UA commands
+
+The Ubuntu Advantage (UA) Tools provides users with a simple mechanism to
```
The rationale here for tools is that the package is called `ubuntu-advantage-tools`, so that's why I named it like this. But we can change it to Client in the docs if we all agree

orndorffgrant replied:
We refer to it as client pretty much everywhere except for the package name I think so far haha :upside_down_face:

lucasmoura commented on docs/tutorials/basic_ua_commands.md:
```diff
@@ -0,0 +1,258 @@
+
+The Ubuntu Advantage (UA) Tools provides users with a simple mechanism to
+view, enable, and disable offerings from Canonical on their system. On this tutorial
+we will cover the base UA commands that allow an user to successfully manage the offering
```
TIL, thanks @orndorffgrant

orndorffgrant commented:
Hi @lucasmoura I should've just merged this when I clicked approve before, but now there is a conflict. Can you fix the conflict and then go ahead and merge?
